ui/pages/confirmations/components/confirm/info/shared/approve-details/approve-details.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  ConfirmInfoRow,
+  ConfirmInfoRowAddress,
+  ConfirmInfoRowDivider,
+} from '../../../../../../../components/app/confirm/info/row';
+import { ConfirmInfoSection } from '../../../../../../../components/app/confirm/info/row/section';
+import { useI18nContext } from '../../../../../../../hooks/useI18nContext';
+import { selectConfirmationAdvancedDetailsOpen } from '../../../../../selectors/preferences';
+import { useDecodedTransactionData } from '../../hooks/useDecodedTransactionData';
+import { Container } from '../transaction-data/transaction-data';
+import {
+  MethodDataRow,
+  OriginRow,
+  RecipientRow,
+} from '../transaction-details/transaction-details';
+
+const Spender = () => {
+  const t = useI18nContext();
+
+  const decodedResponse = useDecodedTransactionData();
+
+  const { value, pending } = decodedResponse;
+
+  if (pending) {
+    return <Container isLoading />;
+  }
+
+  if (!value) {
+    return null;
+  }
+
+  return (
+    <>
+      <ConfirmInfoRow label={t('spender')} tooltip={t('spenderTooltipDesc')}>
+        <ConfirmInfoRowAddress address={value.data[0].params[0].value} />
+      </ConfirmInfoRow>
+
+      <ConfirmInfoRowDivider />
+    </>
+  );
+};
+
+export const ApproveDetails = () => {
+  const showAdvancedDetails = useSelector(
+    selectConfirmationAdvancedDetailsOpen,
+  );
+
+  return (
+    <ConfirmInfoSection>
+      <Spender />
+      <OriginRow />
```

Comment:
Another example here where it would be great to just include our additional `Spender` row but have all the other rows and sections defined once in a single template such as `BaseTransactionInfo`.

Reply:
This is slightly different than what happens in `TransactionDetails` in the contract interaction component because here `RecipientRow` and `MethodDataRow` are shown conditionally with `showAdvancedDetails`. I am not sure we benefit from grouping components further.